datatypes/src/raster/tiling.rs

```diff
@@ -0,0 +1,206 @@
+        GridBoundingBox2D::new_unchecked(start, end)
+    }
+
+    pub fn tile_grid_box(&self, bounding_box: BoundingBox2D) -> GridBoundingBox2D {
+        let start = self.pixel_idx_to_tile_idx(self.upper_left_pixel_idx(bounding_box));
+        let end = self.pixel_idx_to_tile_idx(self.lower_right_pixel_idx(bounding_box));
+        GridBoundingBox2D::new_unchecked(start, end)
+    }
+
+    /// generates the tile idx for the tiles intersecting the bounding box
+    pub fn tile_idx_iterator(
+        &self,
+        bounding_box: BoundingBox2D,
+    ) -> impl Iterator<Item = GridIdx2D> {
+        let GridIdx([upper_left_tile_y, upper_left_tile_x]) =
+            self.pixel_idx_to_tile_idx(self.upper_left_pixel_idx(bounding_box));
+
+        let GridIdx([lower_right_tile_y, lower_right_tile_x]) =
+            self.pixel_idx_to_tile_idx(self.lower_right_pixel_idx(bounding_box));
+
+        let y_range = upper_left_tile_y..=lower_right_tile_y;
+        let x_range = upper_left_tile_x..=lower_right_tile_x;
+
+        y_range.flat_map(move |y_tile| x_range.clone().map(move |x_tile| [y_tile, x_tile].into()))
+    }
+
+    /// generates the tile idx for the tiles intersecting the bounding box
+    pub fn tile_information_iterator(
+        &self,
+        bounding_box: BoundingBox2D,
+    ) -> impl Iterator<Item = TileInformation> {
+        let tile_pixel_size = self.tile_pixel_size;
+        let geo_transform = self.geo_transform;
+        self.tile_idx_iterator(bounding_box)
+            .map(move |idx| TileInformation::new(idx, tile_pixel_size, geo_transform))
+    }
+}
+
+/// The `TileInformation` is used to represent the spatial position of each tile
+#[derive(PartialEq, Debug, Copy, Clone, Serialize, Deserialize)]
+pub struct TileInformation {
+    pub tile_size_in_pixels: GridShape2D,
+    pub global_tile_position: GridIdx2D,
+    pub global_geo_transform: GeoTransform,
+}
+
+impl TileInformation {
+    pub fn new(
+        global_tile_position: GridIdx2D,
+        tile_size_in_pixels: GridShape2D,
+        global_geo_transform: GeoTransform,
+    ) -> Self {
+        Self {
+            global_tile_position,
+            tile_size_in_pixels,
+            global_geo_transform,
+        }
+    }
+
+    #[allow(clippy::unused_self)]
+    pub fn local_upper_left_idx(&self) -> GridIdx2D {
+        [0, 0].into()
+    }
+
+    pub fn local_lower_left_idx(&self) -> GridIdx2D {
+        [self.tile_size_in_pixels.axis_size_y() as isize - 1, 0].into()
+    }
+
+    pub fn local_upper_right_idx(&self) -> GridIdx2D {
+        [0, self.tile_size_in_pixels.axis_size_x() as isize - 1].into()
+    }
+
+    pub fn local_lower_right_idx(&self) -> GridIdx2D {
+        let GridIdx([y, _]) = self.local_lower_left_idx();
+        let GridIdx([_, x]) = self.local_upper_right_idx();
+        [y, x].into()
+    }
+
+    pub fn global_tile_position(&self) -> GridIdx2D {
+        self.global_tile_position
+    }
+
+    pub fn global_upper_left_idx(&self) -> GridIdx2D {
```

Comment:
this function returns the pixel index right? Maybe it should be relfected in the name or at least as a doc coment. Same for the other functions. As both pixels and grids have the same data type for indexes we have to make it clear by name what is meant.

Reply:
1f6f43c